src/cqu_jxgl/student.py

```diff
@@ -57,9 +57,18 @@ def _login(student):
                 headers=headers,
                 proxies=student.proxies,
             )
-            if "账号或密码不正确！请重新输入。" in response.text:
+            if "正在加载权限数据" in response.text:
+                return response, session
+            elif "账号或密码不正确！请重新输入。" in response.text:
                 raise ValidationError("账号或密码不正确！请重新输入。")
-            return response, session
+            elif "账号或密码不正确！请重新输入。" in response.text:
+                raise ValidationError("账号或密码不正确！请重新输入。")
+            elif "该账号尚未分配角色!" in response.text:
+                raise ValidationError("无此帐号。")
+            elif "alert('您尚未报到注册成功，请到学院咨询并办理相关手续！" in response.text:
+                raise ValidationError("此帐号尚未注册成功")
+            else:
+                raise ValidationError("意料之外的登陆返回页面")
 
```

Comment:
I found a better way to do this. It eliminates the "elif" mess. What do you say?
```
from bs4 import BeautifulSoup as BS
soup = BS(response.content)
result = soup.find("span", id="divLogNote").text
if "正在加载权限数据" in result:
                 return response, session
else raise ValidationError(result)
```

Reply:
Great idea!
But '您尚未报到注册成功' is a special case where the error information is in the javascript code.
I can provide the html file later today if I can find it ^_^

Reply:
It's totally fine. Just add an elif and commit it.

Reply:
Is it ok to add such a dependency BeautifulSoup just for these cases?

Reply:
I don't see a problem here. All packages depend on cqu-jxgl depend on bs4 anyway.

Reply:
`Re` may also works fine, it can also be used to filter out the alert statements.

@0xSeanll is a fan of bs, lol

Both way lgtm, choose the one you prefer :)